src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -232,6 +230,34 @@ private async Task<SearchResult> SearchImpl(SearchOptions searchOptions, bool hi
                             rawResource = await streamReader.ReadToEndAsync();
                         }
 
+                        // if we have more columns, it means sort expressions were added.
+                        if (reader.FieldCount > 9)
```

Comment:
We only need this if it's the last match row (See `if (matchCount >= searchOptions.MaxItemCount && isMatch)` above).

Reply:
(or it was above, at least)

Reply:
@johnstairs ,
`reader.FieldCount` is about how many **columns** we got back. More than 9 means, we have `SortExpr` columns which we need to use.
However, `matchCount` and `searchOptions.MaxItemCount` relate to the **rows**, which has nothing to do with the `SortExpr` value that we are after - in this section.
Therefore, I am not sure there is much to change

Reply:
I am saying that you only need to read the sortexpr column if it's the last match row and it has exceeded the row count.

Reply:
Done